### ISSUE TITLE
synapse will not start if services.matrix-synapse.settings.trusted_key_servers does not specify verify_keys

### ISSUE THREAD

**Issue by lunik1** [NixOS/nixpkgs#163669]
### Describe the bug
Not specifying `verify_keys` in any of the  `services.matrix-synapse.settings.trusted_key_servers` submodules will cause synapse to fail to start. This setting is specified as optional by usptream https://matrix-org.github.io/synapse/latest/usage/configuration/homeserver_sample_config.html.

Error message:
```
[2541475.987207] synapse_homeserver[299]: Error in configuration:
[2541475.987473] synapse_homeserver[299]:   Unable to parse 'trusted_key_servers': None is not of type 'object'
```

### Steps To Reproduce
1. Set `services.matrix-synapse.settings.trusted_key_servers` to e.g.
```nix
trusted_key_servers = [
  { server_name = "matrix.org"; }
];
```
2. Start the synapse service

### Expected behavior
Synapse should start

### Additional context
Workaround: set `verify_keys` to an empty set.

### Notify maintainers

@Ma27 @fadenb @mguentner @Ekleog @Ralith @dandellion @sumnerevans @mweinelt

**Comment by mweinelt** [comment]
We set a sane default in the module, so not overwriting it will work.

https://github.com/NixOS/nixpkgs/blob/cfdda97302f20c64d0f1055c40ebf4e9724fa137/nixos/modules/services/matrix/matrix-synapse.nix#L657-L662

We could in theory add an assertion for that, but I lack the motivation for that corner case.

**Comment by lunik1** [comment]
This also happens if you add a trusted key server that isn't matrix.org (which is advisable, to reduce centralisation). From looking at the code in https://github.com/matrix-org/synapse/blob/develop/synapse/config/key.py I think the empty set has the same effect, so can this not be made the default for `verify_keys`?

**Comment by mweinelt** [comment]
So what you are saying is that we should make it default to an empty attrset instead of null?

**Comment by lunik1** [comment]
Yes. Ideally if not set the resulting config file would omit the `verify_keys` key entirely, but I'm not sure how easy that would be. An empty attrset is a fine compromise, as it seems to have the same effect.

**Comment by emilylange** [comment]
Fixed by the now long merged #232837
